Migrate dependencies for watsonx.data

iamar7 opened the pull request:
### Description
Migration for Watsonx-data
<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

iamar7 commented:
/run pipeline

iamar7 commented:
/run pipeline

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -21,20 +21,21 @@
       ],
       "short_description": "Creates and configures an instance of IBM watsonx.data.",
-      "long_description": "This architecture supports creating and configuring an instance of watsonx.data.",
+      "long_description": "This architecture supports creating and configuring an instance of watsonx.data. For more details on watsonx data, [see here](https://cloud.ibm.com/docs/watsonxdata/index.html).",
```
Good to have one or two lines for watsonx.data. You can use its documentation. Idea here is  to show the value of your DA then only it will be used. Also, how it is different than the catalog's watsonx.data tile.

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -21,20 +21,21 @@
       "provider_name": "IBM",
       "features": [
         {
           "title": "Creates an instance of IBM watsonx.data",
-          "description": "Creates and configures an IBM watsonx.data instance."
+          "description": "For more details on Watsonx Data instances, [see here](https://cloud.ibm.com/watsonxdata#about)"
         },
         {
           "title": "Creates a KMS encryption enabled IBM watsonx.data instance",
           "description": "Create and configures a KMS encryption enabled IBM watsonx.data instance for Enterprise plan."
         }
```
* Optional integration is missing. Already mentioned [here](https://github.ibm.com/GoldenEye/issues/issues/13592#issuecomment-112085880)
* The first point can be deleted as per previous discussion. Initiated the thread in cpre and lets see whats the final outcome.
* `see here` can be written as `refer here` / `refer this` / `learn more`

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -21,20 +21,21 @@
           "description": "Create and configures a KMS encryption enabled IBM watsonx.data instance for Enterprise plan."
         }
       ],
+      "support_details": "This product is in the community registry, as such support is handled through the originated repo. If you experience issues please open an issue in that repository [https://github.com/terraform-ibm-modules/terraform-ibm-watsonx-data/issues](https://github.com/terraform-ibm-modules/terraform-ibm-watsonx-data/issues). Please note this product is not supported via the IBM C ...
```
Do not need to show the full link (already discussed in DA rallies). Please hyperlink. Refer [this](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc/blob/issue_13328/ibm_catalog.json#L74)

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,9 +54,26 @@
               ],
-              "service_name": "all-account-management-services"
+              "service_name": "kms",
+              "notes": "Required if you are creating a new Key Protect instance or provisioning associated resources like key rings and keys."
```
`[Optional]` prefix is missing.
I'll prefer to write - `Required if KMS encryption is enabled and Key protect is used for encryption.`

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,9 +54,26 @@
+                "crn:v1:bluemix:public:iam::::role:Editor"
+              ],
+              "service_name": "hs-crypto",
+              "notes": "Required if you are provisioning resources like key rings and keys for an existing Hyper Protect Crypto Services (HPCS) instance."
```
same as above.

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,9 +54,26 @@
+                "crn:v1:bluemix:public:iam::::role:Administrator"
+              ],
+              "service_name": "iam-identity",
+              "notes": "Required if Cloud automation for account configuration is enabled."
```
[Optional] prefix is missing

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -67,18 +85,22 @@
               {
-                "title": " Creates and configures an instance of watsonx.data",
-                "description": "This architecture creates and configures an instance of IBM watsonx.data. It accepts or creates a resource group."
+                "title": "Watsonx data instance",
+                "description": "Creates and configures an instance of watsonx.data."
```
Not required as variation feature, already covered in Overview section.

iamar7 replied:
Do we remove the feature section from architecture entirely?

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -67,18 +85,22 @@
+                "description": "Creates and configures an instance of watsonx.data."
+              },
+              {
+                "title": "Key Rings and Keys",
+                "description": "Creates Key Rings and Keys for either watsonx.data instances."
```
`for either watsonx.data instances` -> need to reword.
`creates key...` -> supports ....

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -67,18 +85,22 @@
               {
                 "diagram": {
-                  "caption": "watsonx.data on IBM Cloud",
+                  "caption": "IBM Watsonx.data architecture",
```
Watsonx -> wrong way to write. It should be `watsonx`
Previous caption was correct.

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -67,18 +85,22 @@
                   "type": "image/svg+xml"
                 },
-                "description": "This architecture supports creating and configuring an instance of watsonx.data."
+                "description": "This architecture supports creating and configuring an instance of IBM watsonx.data resources."
```
Need to add more details as discussed already. 
https://github.ibm.com/GoldenEye/issues/issues/13592#issuecomment-112084362

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -223,44 +232,39 @@
-              "on_by_default": true,
-              "version": "^v2.0.6"
+              "on_by_default": false,
+              "version": "^v3.0.7"
```
Need to pin the version

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -130,20 +152,7 @@
-                  "value": "public-and-private"
-                }
-              ]
+              "hidden": true
```
We plan not to hide now.

iamar7 replied:
Do we plan to hide this now?

maheshwarishikha commented:
For architecture diagram, missing points are:
* optional prefix is missing for addons
* No connected arrow is shown with Observability
* Optional components can be in different resource group so move out from Resource group box
* Check the names of the services. Please use official names

iamar7 commented:
/run pipeline

maheshwarishikha commented:
* Please update the Observability in the diagram as discussed [here](https://ibm-cloudplatform.slack.com/archives/CG06FQ31P/p1747999880519089?thread_ts=1747813475.750729&cid=CG06FQ31P)
* Also, make doc changes as discussed.

imprateeksh commented on ibm_catalog.json:
```diff
@@ -130,20 +157,7 @@
             },
             {
               "key": "provider_visibility",
```
Do not remove the options from here.

imprateeksh reviewed (dismissed):
@iamar7 Please check the comments and diagram should be revisited - 
 - Two boxes have blue bar on left means this is some other Component, so prefer to use rectangle as there is no Observability or Optional component box.
 - [Optional] should be the name of outer box.
 - Observability should not have `[Optional]` as it is already part of Optional box.
 - It would be good if the Observability resources are arranged alphabetically.

iamar7 commented:
/run pipeline

imprateeksh commented on ibm_catalog.json:
```diff
@@ -53,32 +58,54 @@
                 "crn:v1:bluemix:public:iam::::role:Editor"
               ],
-              "service_name": "lakehouse"
+              "service_name": "lakehouse",
```
Yes I get it but even I am doubtful of this point. So let's confirm this.

maheshwarishikha replied:
Its ok, we can keep this.

daniel-butler-irl commented on ibm_catalog.json:
```diff
@@ -21,20 +21,25 @@
       ],
       "short_description": "Creates and configures an instance of IBM watsonx.data.",
-      "long_description": "This architecture supports creating and configuring an instance of watsonx.data.",
+      "long_description": "IBM® watsonx.data is a data management solution for collecting, storing, querying, and analyzing all your enterprise data with a single unified data platform. The Cloud automation for watsonx.data allows user to set up an instance of watsonx.data. For more details on watsonx.data, [refer here](https://cloud.ibm.com/docs/watsonxdata/index.html).",
```
Unless we where asked to add it remove `®`

iamar7 commented on ibm_catalog.json:
```diff
@@ -223,59 +237,48 @@
+                  "version_input": "enable_kms_encryption",
+                  "value": true
                 },
                 {
```
Added

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,32 +58,54 @@
+                "title": "watsonx.data instance",
+                "description": "Creates and configures an instance of watsonx.data."
+              },
+              {
```
Lets keep as `[Optional]` only. Also, please make the outer box as dotted lines box.

iamar7 commented:
/run pipeline

daniel-butler-irl reviewed (dismissed):
Looks good to me, @maheshwarishikha  could you give the final approval and merge if you are happy. Thanks.

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,32 +58,54 @@
               {
                 "diagram": {
-                  "caption": "watsonx.data on IBM Cloud",
+                  "caption": "IBM watsonx.data architecture",
```
Please revert the caption, as such we are really not talking about watsonx.data architecture.

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -53,22 +58,44 @@
               ],
-              "service_name": "lakehouse"
+              "service_name": "hs-crypto",
+              "notes": "[Optional] Required if Hyper Protect Crypto Services (HPCS) encryption is enabled and is used for encryption."
```
Description should be changed to -
```
[Optional] Required if KMS encryption is enabled and Hyper Protect Crypto Services is used for encryption.
```

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -223,59 +237,53 @@
-              "name": "testing-deploy-arch-ibm-kms",
-              "id": "85fdbd7a-8c77-4abd-b716-653a31f3aba9-global",
-              "version": "^1.0.6",
+              "name": "deploy-arch-ibm-kms",
```
I would suggest to include region variable in input mapping of KMS.

```
{
                  "dependency_input": "region",
                  "version_input": "region",
                  "reference_version": true
                }
```

iamar7 commented:
> @iamar7 Please recheck the diagram carefully:
> 
> * I do see the difference in fonts for Observability icons Cloud logs and Cloud Monitoring. Please check.
> * You changed the arrow after our discussion, but why there is a difference in arrows' thickness?
> * Also, can you please make `watsonx.data` icon little smaller and move into centre? I think..that alignment will be better.

Resolved the fonts difference. Updated the arrow and moved the watsonx.data to the center and made it smaller.

iamar7 commented:
/run pipeline

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -290,18 +298,16 @@
-              "id": "daf2cf91-2384-4205-b7f4-621299a4896d-global",
-              "version": "^1.3.0",
+              "name": "deploy-arch-ibm-observability",
+              "description": "Enable to provision and configure IBM Cloud Logs, Cloud Monitoring, Metrics routing and Activity Tracker event routing for analysing logs and metrics generated by the watsonx.data instance. ",
```
As discussed in yesterday's DA rally,
`Enable to provision and configure` -> Configure
Also we can remove `Metrics routing` as not shown in diagram too.

maheshwarishikha commented:
Update the diagram as discussed here:
 https://ibm-cloudplatform.slack.com/archives/CG06FQ31P/p1747999880519089?thread_ts=1747813475.750729&cid=CG06FQ31P

iamar7 commented:
/run pipeline
